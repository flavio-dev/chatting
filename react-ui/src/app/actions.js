export const SET_LIST_USERS = 'SET_LIST_USERS'
export const GET_INITIAL_LIST_USERS = 'GET_INITIAL_LIST_USERS'

export const setListUsers = (listUsers) => {
  return {
    type: SET_LIST_USERS,
    listUsers
  }
}

export const getInitialListUsers = () => {
  return {
    type: GET_INITIAL_LIST_USERS
  }
}
