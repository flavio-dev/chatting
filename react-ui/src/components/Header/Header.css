.Header {
  color: #00A0CC;
  background-color: #F4F9FF;
  font-family: "Lobster", Helvetica, sans-serif;
  border-bottom: 5px solid #B78686; }

.HeaderContent {
  margin-left: auto;
  margin-right: auto;
  width: 95%;
  align-items: center;
  display: flex;
  height: 80px; }
  @media only screen and (min-width: 601px) {
    .HeaderContent {
      width: 90%; } }

.HeaderLogo {
  height: 56px;
  transform: rotate(72deg);
  width: 56px; }

.HeaderTitle {
  font-size: 40px;
  margin-left: 30px; }
