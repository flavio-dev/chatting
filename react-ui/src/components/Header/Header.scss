@import "globals.scss";

.Header {
  color: $color-blue-sky;
  background-color: $color-white;
  font-family: $title-font-family;
  border-bottom: 5px solid $color-milk-chocolate;
}

.HeaderContent {
  @include default-wrapper-width();
  align-items: center;
  display: flex;
  height: 80px;
}

.HeaderLogo {
  height: 56px;
  transform: rotate(72deg);
  width: 56px;
}

.HeaderTitle {
  font-size: 40px;
  margin-left: 30px;
}
