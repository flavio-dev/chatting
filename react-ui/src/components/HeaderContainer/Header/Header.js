import React from 'react'

import styles from './Header.css'
import logo from './logo.png'

export const Header = ({disconnecting, userId, connectionStatus}) => (
  <div className={styles.Header}>
    <div className={styles.HeaderContent}>
      <div className={styles.HeaderBrand} onClick={() => disconnecting(userId)}>
        <img src={logo} className={styles.HeaderLogo} alt='logo' />
        <div className={styles.HeaderTitle}>Chatting</div>
      </div>
      {connectionStatus === 'connected' &&
        <button onClick={() => disconnecting(userId)}>Logout</button>
      }
    </div>
  </div>
)

export default Header
