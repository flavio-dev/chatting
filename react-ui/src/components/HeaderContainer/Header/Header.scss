@import "globals.scss";

.Header {
  background-color: $color-white;
  border-bottom: 5px solid $color-milk-chocolate;
  color: $color-blue-sky;
  font-family: $title-font-family;
}

.HeaderContent {
  @include default-wrapper-width();
  align-items: center;
  display: flex;
  height: 80px;
  justify-content: space-between;
}

.HeaderBrand {
  align-items: center;
  display: flex;
}

.HeaderLogo {
  height: 56px;
  transform: rotate(72deg);
  width: 56px;
}

.HeaderTitle {
  font-size: 40px;
  margin-left: 30px;
}
