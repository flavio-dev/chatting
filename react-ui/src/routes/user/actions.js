export const SET_USER_ID = 'SET_USER_ID'
export const SELECT_USER = 'SELECT_USER'

export const setUserId = (userId) => {
  return {
    type: SET_USER_ID,
    userId
  }
}

export const selectUser = (userId) => {
  return {
    type: SELECT_USER,
    userId
  }
}
