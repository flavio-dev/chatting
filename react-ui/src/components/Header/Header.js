import React from 'react'
import { connect } from 'react-redux'
import { push } from 'react-router-redux'

import { disconnecting } from 'store/actions'

import styles from './Header.css'
import logo from './logo.png'

export const Header = ({disconnecting}) => (
  <div className={styles.Header}>
    <div className={styles.HeaderContent}>
      <div className={styles.HeaderBrand}>
        <img src={logo} className={styles.HeaderLogo} alt='logo' />
        <div className={styles.HeaderTitle} onClick={disconnecting}>Chatting</div>
      </div>
      <button onClick={disconnecting}>Logout</button>
    </div>
  </div>
)

const mapActionCreators = (dispatch) => ({
  disconnecting: () => {
    dispatch(disconnecting())
    dispatch(push('/'))
  }
})

const mapStateToProps = (state) => {}


export default connect(mapStateToProps, mapActionCreators)(Header)
