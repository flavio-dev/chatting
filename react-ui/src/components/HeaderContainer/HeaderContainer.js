import { connect } from 'react-redux'
import { push } from 'react-router-redux'

import { disconnecting } from 'store/actions'
import { getUserId } from 'routes/user/selectors'
import { getConnectionState } from 'app/selectors'

import Header from './Header'

const mapActionCreators = (dispatch) => ({
  disconnecting: (userId) => {
    dispatch(disconnecting(userId))
    dispatch(push('/'))
  }
})

const mapStateToProps = (state) => ({
  userId: getUserId(state),
  connectionStatus: getConnectionState(state)
})

export default connect(mapStateToProps, mapActionCreators)(Header)
