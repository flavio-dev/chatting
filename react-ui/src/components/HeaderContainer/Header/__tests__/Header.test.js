import React from 'react'
import Immutable from 'immutable'
import Header from '../Header'
import HeaderContainer from '../../HeaderContainer'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import { disconnecting as disconnectingAction } from 'store/actions'
import configureMockStore from 'redux-mock-store'

describe('Testing Header.js', () => {
  it('renders correctly with logout button', () => {
    const disconnecting = jest.fn()
    const tree = renderer
      .create(<Header disconnecting={disconnecting} userId='userId' connectionStatus='connected' />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly without logout button', () => {
    const disconnecting = jest.fn()
    const tree = renderer
      .create(<Header disconnecting={disconnecting} userId='userId' connectionStatus='disconnected' />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the logout button is connected', () => {
    const disconnecting = jest.fn()
    const wrapper = mount(
      <Header disconnecting={disconnecting} userId='userId' connectionStatus='connected' />
    )

    expect(wrapper.find('button').length).toBe(1)
  })

  it('hides the logout button is disconnected', () => {
    const disconnecting = jest.fn()
    const wrapper = mount(
      <Header disconnecting={disconnecting} userId='userId' connectionStatus='disconnected' />
    )

    expect(wrapper.find('button').length).toBe(0)
  })

  it('triggers an function passing userId as param when clicking on the button', () => {
    const disconnecting = jest.fn()
    const wrapper = mount(
      <Header disconnecting={disconnecting} userId='userId' connectionStatus='connected' />
    )
    expect(disconnecting.mock.calls.length).toBe(0)
    wrapper.find('button').simulate('click')
    expect(disconnecting.mock.calls.length).toBe(1)
    expect(disconnecting.mock.calls[0][0]).toBe('userId')
  })

  it('triggers the DISCONNECTED action', () => {
    const mockStore = configureMockStore();
    const store = mockStore(
      Immutable.Map({
        appReducers: Immutable.Map({
          connectionStatus: 'connected'
        }),
        userReducers: Immutable.Map({
          userId: 'userId'
        })
      })
    )

    const wrapper = mount(
      <HeaderContainer store={store} />
    )

    wrapper.find('button').simulate('click')
    const actions = store.getActions()
    expect(actions[0]).toEqual(disconnectingAction('userId'))
  })
})
