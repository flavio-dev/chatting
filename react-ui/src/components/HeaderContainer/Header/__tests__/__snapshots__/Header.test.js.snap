// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header.js renders correctly with logout button 1`] = `
<div
  className="Header"
>
  <div
    className="HeaderContent"
  >
    <div
      className="HeaderBrand"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="HeaderLogo"
        src="logo.png"
      />
      <div
        className="HeaderTitle"
      >
        Chatting
      </div>
    </div>
    <button
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
</div>
`;

exports[`Testing Header.js renders correctly without logout button 1`] = `
<div
  className="Header"
>
  <div
    className="HeaderContent"
  >
    <div
      className="HeaderBrand"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="HeaderLogo"
        src="logo.png"
      />
      <div
        className="HeaderTitle"
      >
        Chatting
      </div>
    </div>
  </div>
</div>
`;
